reckoner/chart.py

```diff
@@ -191,63 +185,168 @@ def install(self, namespace=None, context=None):
             self.args.append('--kube-context={}'.format(self.context))
+
+        # Add debug arguments
         self.args.extend(self.debug_args)
+
+        # Add the version arguments
         if self.version:
             self.args.append('--version={}'.format(self.version))
-        for file in self.files:
-            self.args.append("-f={}".format(file))
-
-        for key, value in self.values.iteritems():
-            for k, v in self._format_set(key, value):
-                self.args.append("--set={}={}".format(k, v))
 
+        # HACK: This is in place until we can fully deprecate the usage of set
+        #       in place of values. Currently values: gets translated into
+        #       --set commands to the helm command line run. This should be
+        #       changed so that set: goes to command line args and values: goes
+        #       to temporary values files, to keep strong types in yaml.
+        #       This will go away once we fully change the functionality of
+        #       values: vs sets: settings in course.yml
+        self._merge_set_and_values()
+
+        # Build the list of existing yaml files to use for the helm command
+        self.build_files_list()
+
+        # Build the file arguments from the `values: {}` in course.yml
+        self.build_temp_values_files()
+
+        # Build the list of --set arguments
+        self.build_set_arguments()
+
+        # Build the list of --set-string arguments
+        self.build_set_string_arguments()
+
+    def _merge_set_and_values(self):
+        '''
+        This is a temporary method that will be gone once the values: vs set:
+        debacle has been resolved. (See https://github.com/reactiveops/reckoner/issues/7)
+
+        NOTE ONLY RUN THIS ONCE BECAUSE IT'S NOT IDEMPOTENT
+        '''
+        if self._hack_set_values_already_merged:
+            raise StandardError('This method cannot be called twice. '
+                                'If you are seeing this please open an '
+                                'issue in github.')
+
+        def merge_dicts(values, sets):
+            '''This does a dict merge and prefers "sets" values'''
+            new_dict = values.copy()
+            new_dict.update(sets)
+            return new_dict
+
+        logging.debug('Merging values: into sets: - sets: take precedence.')
+        logging.debug('Original value of sets: {}'.format(self.set_values))
+        self.set_values = merge_dicts(self.values, self.set_values)
+        logging.debug('New value of sets: {}'.format(self.set_values))
+        self._hack_set_values_already_merged = True
+
+    def build_temp_values_files(self):
+        '''
+        This method builds temporary files based on the values: settings
+        provided in the course.yml. This effectively keeps type persistence
+        between the course.yml and what is passed to helm. If you use set:
+        value arguments then you can lose types like int, float and true/false
+        '''
```

Comment:
nit: several places use `'''` instead of `"""`.  Not sure if intentional.

Reply:
So the `"""` are from code i'd moved around - I had been using `'''` - I'd actually not notices that I had used the opposite that the other files had used. I can refactor but there [isn't really any difference](https://stackoverflow.com/questions/7783097/when-to-use-triple-single-quotes-instead-of-triple-double-quotes) in the implementation of what it does.

Reply:
As long as we are consistent, I don't care, and I know it's not different.  Thus the `nit` :-D